tests/test_activate.py

```diff
@@ -3355,6 +3357,126 @@ def test_activate_and_deactivate_for_uninitialized_env(conda_cli):
     )
 
 
+def test_keep_case(monkeypatch: MonkeyPatch):
+    monkeypatch.setenv("CONDA_KEEP_CASE", True)
+    reset_context()
+    assert context.keep_case
+
+    activator = PosixActivator()
+    export_vars, unset_vars = activator.get_export_unset_vars(
+        one=1,
+        TWO=2,
+        three=None,
+        FOUR=None,
+    )
+
+    # original lowercase
+    assert "one" in export_vars
+    assert "three" in unset_vars
+
+    # to uppercase
+    assert "ONE" in export_vars
+    assert "THREE" in unset_vars
+
+    # original uppercase
+    assert "TWO" in export_vars
+    assert "FOUR" in unset_vars
+
+
+def test_not_keep_case(monkeypatch: MonkeyPatch):
+    monkeypatch.setenv("CONDA_KEEP_CASE", False)
+    reset_context()
+    assert not context.keep_case
+
+    activator = PosixActivator()
+    export_vars, unset_vars = activator.get_export_unset_vars(
+        one=1,
+        TWO=2,
+        three=None,
+        FOUR=None,
+    )
+
+    # original lowercase
+    assert "one" not in export_vars
+    assert "three" not in unset_vars
+
+    # to uppercase
+    assert "ONE" in export_vars
+    assert "THREE" in unset_vars
+
+    # original uppercase
+    assert "TWO" in export_vars
+    assert "FOUR" in unset_vars
```

Comment:
Combine both of these with parameterization:

```suggestion
@pytest.mark.parametrize("keep_case", [True, False])
def test_keep_case(monkeypatch: MonkeyPatch, keep_case: bool):
    monkeypatch.setenv("CONDA_KEEP_CASE", keep_case)
    reset_context()
    assert context.keep_case is keep_case

    activator = PosixActivator()
    export_vars, unset_vars = activator.get_export_unset_vars(
        one=1,
        TWO=2,
        three=None,
        FOUR=None,
    )

    # original lowercase (conditionally true per keep_case)
    assert ("one" in export_vars) is keep_case
    assert ("three" in unset_vars) is keep_case

    # to uppercase (always true)
    assert "ONE" in export_vars
    assert "THREE" in unset_vars

    # original uppercase (always true)
    assert "TWO" in export_vars
    assert "FOUR" in unset_vars
```